src/components/popover/popover.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+    Component,
+    h,
+    Prop,
+    Element,
+    Event,
+    EventEmitter,
+} from '@stencil/core';
+import { createRandomString } from '../../util/random-string';
+
+/**
+ * @slot - header - Content to put inside the header surface
+ * @slot - body - Content to put inside the body surface
+ * @slot - footer - Content to put inside the footer surface
+ * @exampleComponent limel-example-popover
+ */
+
+@Component({
+    tag: 'limel-popover',
+    shadow: true,
+})
+export class Popover {
+    /**
+     * True if the content within the popover should be visible
+     */
+    @Prop()
+    public visible = false;
+
+    /**
+     * Emits an event when the component is closing
+     */
+    @Event()
+    private close: EventEmitter<void>;
+
+    @Element()
+    private host: HTMLLimelPopoverElement;
+
+    private portalId: string;
+
+    constructor() {
+        this.portalId = createRandomString();
+        this.globalClickListener = this.globalClickListener.bind(this);
+    }
+
+    public componentDidLoad() {
+        document.addEventListener('click', this.globalClickListener, {
+            passive: true,
+        });
+    }
+
+    public disconnectedCallback() {
+        document.removeEventListener('click', this.globalClickListener);
+    }
+
+    public render() {
+        return (
+            <limel-portal visible={this.visible} containerId={this.portalId}>
+                <limel-popover-surface
+                    open={this.visible}
+                    contentCollection={this.host.children}
+                >
+                    <slot name="header" />
+                    <slot name="body" />
+                    <slot name="footer" />
+                </limel-popover-surface>
+            </limel-portal>
+        );
+    }
+
+    private globalClickListener({ target }) {
+        const clickedInside = Boolean(target.closest(`#${this.portalId}`));
+        if (this.visible && !clickedInside) {
+            this.visible = false;
```

Comment:
This will not do anything since this prop is not mutable.
However, I think it's a good idea that we set it ourselves and not forcing the user to listen to the `close` event.

I had a thought regarding behaviour similar as this the other day, and I think it could be a good idea to do something like this if we have a "default" behavior, e.g.

```ts
const event = this.close.emit();
if (!event.defaultPrevented) {
    this.visible = false;
}
```

This way we can have a default behavior that will close the popover, and the consumer can opt out of it by calling `event.preventDefault()`. They also still have the possibility to control the state from outside by setting the `visible` prop. We could probably do the same for the dialog and menu components etc. What do you think? :slightly_smiling_face:

Reply:
> This will not do anything since this prop is not mutable.

Are you suggesting that we make this prop mutable or that we add some other state variable to the component?

Reply:
If we are keeping this line we need to make it mutable, otherwise we can just remove it :slightly_smiling_face: 
It depends on if we want to have a default behavior or not

Reply:
Adding another state variable will probably just make it hard to keep in sync with `visible`

Reply:
> I had a thought regarding behaviour similar as this the other day, and I think it could be a good idea to do something like this if we have a "default" behavior, e.g.
> 
> ```ts
> const event = this.close.emit();
> if (!event.defaultPrevented) {
>     this.visible = false;
> }
> ```
> 
> This way we can have a default behavior that will close the popover, and the consumer can opt out of it by calling `event.preventDefault()`. They also still have the possibility to control the state from outside by setting the `visible` prop. We could probably do the same for the dialog and menu components etc. What do you think?

I like your idea!

But on the same time I'm not completely understanding it :)

I thought that the emitted event was fired asynchronously. Thus -> The if statement shouldn't work according to me. (But it seems to do)

Can you explain it for me? :)

Reply:
No I think the events are sent synchronously :thinking: Otherwise this would not work :sweat_smile: 

If we do it like this we can have a default behavior where the popover closes without forcing the user to listen to the `close` event. If they don't want to have the default behavior they can prevent it by invoking `event.preventDefault()`. I think this is a bit more clean than having multiple events like `beforeClose` and stuff like that to prevent some behavior.

On the other hand I'm not sure if it will be possible to open the popover again without listening to the `close` event and updating the local `isOpen` state in the consumer component :thinking:

Reply:
Then everything makes sense! Thanks!🤸‍♂️🌟

Reply:
Seems like it is hard to open it again after it has been closed if we have the default behavior and the consumer is not using a close listener, so I'm not sure there is any benefit to it :thinking: If we would have it like that I think we need to have an imperative API with a method on the element named `open()` that you can invoke to open it, but we don't have methods like that in other components and I believe it's best practice to not have them. I just removed this line for now and maybe we can revisit this feature in the future

Reply:
what about switching name `visible` -> `open`?

Reply:
It has been renamed :slightly_smiling_face:

Reply:
I don’t understand the technical difficulties you are facing. 

But usually consumers don’t care that much about these small details. 

Thus the risk is that they don't do anything about it, and we open several popovers, without closing the previous ones. This messes up the screen. 

I think it’s good to have a default behaviour for clicking outside which is close, and let them override it if they have to. That’s the normal and universal behaviour of popovers.